### PR TITLE
feat(sync-actions): add sync actions for assets

### PR DESCRIPTION
**Pull request by markus-azer**
#### Summary

Generate actions for asset updates 
supported update action :
- setAssetKey
- changeAssetName
- setAssetDescription
- setAssetTags
- setAssetSources
- setAssetCustomType
- setAssetCustomField

### CODE REVIEW

File: packages/sync-actions/src/product-actions.js [hunk 1]
```diff
@@ -381,37 +388,81 @@ function toVariantIdentifier(variant) {
...
+      oldVariant.assets,
+      newVariant.assets
+    )
+    if (REGEX_NUMBER.test(key)) {
+      if (Array.isArray(asset) && asset.length) {
+        addAssetActions.push({
+          action: 'addAsset',
+          asset: diffpatcher.getDeltaValue(asset),
+          ...toVariantIdentifier(newVariant),
+          position: Number(key),
```
Comment (markus-azer) [inline packages/sync-actions/src/product-actions.js:416]: now we also support adding position for the new asset

**Review by daern91 — COMMENTED** [review]
Code looks good to me, looking forward to using this "proper" methodology 👍 I'll just leave a comment for now and cc @tdeekens so FE teams are aware of these changes. 

Not sure if you've been using these sync-actions @tdeekens.

File: packages/sync-actions/src/product-actions.js [hunk 2]
```diff
@@ -532,16 +586,24 @@ export function actionsMapAssets(diff, oldObj, newObj, variantHashMap) {
...
+        const [a, c, r] = _buildVariantAssetsActions(
+          variant.assets,
           oldVariant,
           newVariant
         )
-        if (assetActions) actions = actions.concat(assetActions)
+
+        // add if (assetActions)
+        addAssetActions = addAssetActions.concat(a)
+        changeAssetActions = changeAssetActions.concat(c)
```
[the remark this answers is not in the document]
Comment (tdeekens) [reply packages/sync-actions/src/product-actions.js:601]: It's just my ligatures. Still `a`, `b` and `c` are not the most descriptive names :)

File: packages/sync-actions/src/product-actions.js [hunk 3]
```diff
@@ -41,6 +48,15 @@ export const referenceActionsList = [
  * HELPER FUNCTIONS
  */
 
+const addAction = (key, resource) =>
+  REGEX_NUMBER.test(key) && Array.isArray(resource) && resource.length
+
+const updateAction = (key, resource) =>
+  REGEX_NUMBER.test(key) && Object.keys(resource).length
+
+const removeAction = (key, resource) =>
```
Comment (tdeekens) [inline packages/sync-actions/src/product-actions.js:57]: Nit: Still want to share it. Usually we (MC Team) tried to name functions which return booleans as getIs*. This has the advantage of communicating clearly what is returned while being assignable as isFoo = getIsFoo() without clashing names. 

Take it it leave it 🤡🤠

**Review by tdeekens — APPROVED** [review]

**Review by daern91 — APPROVED** [review]

**Pull request merged**